frontend/src/metabase/search/components/SidebarFilter/SidebarFilter.tsx

```diff
@@ -89,52 +95,60 @@ export const SidebarFilter = ({
   };
 
   return (
-    <div data-testid={dataTestId} ref={dropdownRef}>
-      <div onClick={() => setIsPopoverOpen(!isPopoverOpen)}>
-        <DropdownFilterElement
-          noPadding
-          fieldHasValueOrFocus={fieldHasValue}
-          legend={fieldHasValue ? title : null}
-        >
-          <Group position="apart">
-            {fieldHasValue ? (
-              <DisplayComponent value={value} />
-            ) : (
-              <Group noWrap>
-                <Icon name={iconName} />
-                <Text weight={700}>{title}</Text>
-              </Group>
-            )}
-            <Button
-              style={{ pointerEvents: "all" }}
-              data-testid="sidebar-filter-dropdown-button"
-              compact
-              c="inherit"
-              variant="subtle"
-              onClick={onClearFilter}
-              leftIcon={<Icon name={getDropdownIcon()} />}
-            />
-          </Group>
-        </DropdownFilterElement>
-      </div>
+    <Box
+      data-testid={dataTestId}
+      ref={dropdownRef}
+      onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+      w="100%"
+    >
+      <DropdownFieldSet
+        noPadding
+        legend={fieldHasValue ? title : null}
+        fieldHasValueOrFocus={fieldHasValue}
+      >
+        <Group position="apart" noWrap w="100%">
+          {fieldHasValue ? (
+            <DisplayComponent value={value} />
+          ) : (
+            <Group noWrap>
+              <Icon name={iconName} />
+              <Text weight={700}>{title}</Text>
+            </Group>
+          )}
+          <DropdownClearButton
+            data-testid="sidebar-filter-dropdown-button"
+            compact
+            c="inherit"
+            variant="subtle"
+            onClick={onClearFilter}
+            leftIcon={<Icon name={getDropdownIcon()} />}
+          />
+        </Group>
+      </DropdownFieldSet>
+
       <Popover
         isOpen={isPopoverOpen}
         onClose={onPopoverClose}
         target={dropdownRef.current}
         ignoreTrigger
         autoWidth
+        sizeToFit
       >
-        <Box p="md" w={popoverWidth ?? "100%"}>
-          <ContentComponent
-            value={selectedValues}
-            onChange={selected => setSelectedValues(selected)}
-          />
-        </Box>
-        <DropdownApplyButtonDivider />
-        <Group position="right" align="center" px="sm" pb="sm">
-          <Button onClick={onApplyFilter}>{t`Apply filters`}</Button>
-        </Group>
+        <SearchEventSandbox>
+          <FocusTrap active>
```

Comment:
The first time I open the popover, the "Apply filters" button is focused.
The second time I open the popover, the "search for users" input is focused.